src/ast.rs

```diff
@@ -0,0 +1,92 @@
+//! Provides a type system for the AST, in some sense
+
+mod expr_ext;
+mod node_ext;
+mod nodes;
+mod operators;
+mod token_ext;
+mod tokens;
+
+use std::marker::PhantomData;
+
+use crate::{
+    NixLanguage,
+    SyntaxKind::{self, *},
+    SyntaxNode, SyntaxNodeChildren, SyntaxToken,
+};
+
+pub use nodes::*;
+pub use operators::{BinOpKind, UnaryOpKind};
+pub use tokens::*;
+
+pub trait AstNode: rowan::ast::AstNode<Language = NixLanguage> {}
+
+impl<T> AstNode for T where T: rowan::ast::AstNode<Language = NixLanguage> {}
+
+#[derive(Debug, Clone)]
+pub struct AstNodeChildren<N> {
+    inner: SyntaxNodeChildren,
+    _p: PhantomData<N>,
+}
```

Comment:
This is [`rowan::ast::AstChildren`](https://docs.rs/rowan/0.15.6/rowan/ast/struct.AstChildren.html), which is returned by [`rowan::ast::support::children`](https://docs.rs/rowan/0.15.6/rowan/ast/support/fn.children.html)